tests/components/withings/conftest.py

```diff
@@ -0,0 +1,379 @@
+        assert await async_setup_component(
+            hass,
+            http.DOMAIN,
+            config.hass_config
+        )
+        assert await async_setup_component(
+            hass,
+            api.DOMAIN,
+            config.hass_config
+        )
+
+        nokia_auth_get_credentials_patch = asynctest.patch(
+            'nokia.NokiaAuth.get_credentials',
+            return_value=nokia.NokiaCredentials(
+                access_token='my_access_token',
+                token_expiry=time.time() + 600,
+                token_type='my_token_type',
+                refresh_token='my_refresh_token',
+                user_id='my_user_id',
+                client_id=config.withings_config.get(const.CLIENT_ID),
+                consumer_secret=config.withings_config.get(const.CLIENT_SECRET)
+            )
+        )
+        nokia_auth_get_credentials_mock = \
+            nokia_auth_get_credentials_patch.start()
+
+        nokia_api_request_patch = asynctest.patch(
+            'nokia.NokiaApi.request',
+            return_value=config.nokia_request_response
+        )
+        nokia_api_request_mock = nokia_api_request_patch.start()
+
+        nokia_api_get_measures_patch = asynctest.patch(
+            'nokia.NokiaApi.get_measures',
+            return_value=config.nokia_measures_response
+        )
+        nokia_api_get_measures_mock = \
+            nokia_api_get_measures_patch.start()
+
+        nokia_api_get_sleep_patch = asynctest.patch(
+            'nokia.NokiaApi.get_sleep',
+            return_value=config.nokia_sleep_response
+        )
+        nokia_api_get_sleep_mock = \
+            nokia_api_get_sleep_patch.start()
+
+        nokia_api_get_sleep_summary_patch = asynctest.patch(
+            'nokia.NokiaApi.get_sleep_summary',
+            return_value=config.nokia_sleep_summary_response
+        )
+        nokia_api_get_sleep_summary_mock = \
+            nokia_api_get_sleep_summary_patch.start()
+
+        data_manager_get_throttle_interval_patch = asynctest.patch(
+            'homeassistant.components.withings.common.WithingsDataManager'
+            '.get_throttle_interval',
+            return_value=config.throttle_interval
+        )
+        data_manager_get_throttle_interval_mock = \
+            data_manager_get_throttle_interval_patch.start()
+
+        patches.extend([
+            nokia_auth_get_credentials_patch,
+            nokia_api_request_patch,
+            nokia_api_get_measures_patch,
+            nokia_api_get_sleep_patch,
+            nokia_api_get_sleep_summary_patch,
+            data_manager_get_throttle_interval_mock
+        ])
+
+        # Collect the flow id.
+        tasks = []
+
+        orig_async_create_task = hass.async_create_task
+
+        def create_task(*args):
+            task = orig_async_create_task(*args)
+            tasks.append(task)
+            return task
+
+        async_create_task_patch = asynctest.patch.object(
+            hass,
+            'async_create_task',
```

Comment:
I'm a bit hesitant to patch `hass.async_create_task`. Can we find a way without patching the core of home assistant?

Eg in the smartthings tests I think we mock the config entry and forward the entry to the platform directly from the component without calling `async_setup_component` directly. Maybe something like that is possible?

https://github.com/home-assistant/home-assistant/blob/94a2fd542ea47abf59d2e3a21f17637a3d9af92f/tests/components/smartthings/conftest.py#L27-L45

Reply:
This patch is only used once (as a wrapper) to retrieve the flow_id of the config flow and then it's released. I experimented with better ways to retrieve the flow_id but had no success. Do you have suggestions?

Reply:
Do we need the flow id specifically or is that just to be able to access something else? In the example above the entry is created inside the test so we have total control of what is forwarded.

Reply:
The flow_id is used later to run through the config flow process by calling `hass.config_entries.flow.async_configure` on conftest.py:222 and other places in conftest.